platforms/ide/ide-plugins/src/main/java/org/gradle/plugins/ide/internal/tooling/IdeaModuleBuilderSupport.java

```diff
@@ -84,28 +88,32 @@ public static List<DefaultIdeaDependency> buildDependencies(Set<Dependency> reso
         for (Dependency dependency : resolvedDependencies) {
             if (dependency instanceof SingleEntryModuleLibrary) {
                 SingleEntryModuleLibrary d = (SingleEntryModuleLibrary) dependency;
-                DefaultIdeaSingleEntryLibraryDependency defaultDependency = new DefaultIdeaSingleEntryLibraryDependency()
-                    .setFile(d.getLibraryFile())
-                    .setSource(d.getSourceFile())
-                    .setJavadoc(d.getJavadocFile())
-                    .setScope(new DefaultIdeaDependencyScope(d.getScope()))
-                    .setExported(d.isExported());
-
-                if (d.getModuleVersion() != null) {
-                    defaultDependency.setGradleModuleVersion(new DefaultGradleModuleVersion(d.getModuleVersion()));
-                }
-                dependencies.add(defaultDependency);
+                dependencies.add(ideaSingleEntryLibraryDependencyFor(d));
             } else if (dependency instanceof ModuleDependency) {
-                ModuleDependency moduleDependency = (ModuleDependency) dependency;
-
-                DefaultIdeaModuleDependency ideaModuleDependency = new DefaultIdeaModuleDependency(moduleDependency.getName())
-                    .setExported(moduleDependency.isExported())
-                    .setScope(new DefaultIdeaDependencyScope(moduleDependency.getScope()));
-
-                dependencies.add(ideaModuleDependency);
+                ModuleDependency d = (ModuleDependency) dependency;
+                dependencies.add(ideaModuleDependencyFor(d));
             }
         }
         return dependencies;
     }
 
+    private static DefaultIdeaSingleEntryLibraryDependency ideaSingleEntryLibraryDependencyFor(SingleEntryModuleLibrary d) {
```

Comment:
It is a private static method, but `d` may still be too short of a name. I think in the Gradle codebase, we at least give one word names to the parameters. Though, I don't feel strongly about it.